src/newtype_macros.rs

```diff
@@ -54,7 +56,9 @@ This function panics if `value` is greater than ", $max, "."
                 pub fn new(value: $repr) -> $name {
                     assert!(
                         $name::is_valid(value),
-                        format!("{} is not a valid value", value)
+                        // requires std
+                        //format!("{} is not a valid value", value)
+                        "Not a valid value"
```

Comment:
@perlindgren Just like you I think that calling `new` with an invalid value is a usage fault and therefore justifies a panic. If anyone doesn't like the possibility of a panic and prefers a `Result`, then they could always use `.try_into()` which is also implemented for each newtype. Therefore I'm glad that the signature of `new` stays the same.

Concerning the removal of `format!`, maybe it would be a good idea to use it if `std` is available and not use it in `no_std` context. In `std` context, it's sometimes very helpful for error analysis to see the actual value that failed when getting a backtrace from production (it recently helped me to fix some user-reported bugs in [ReaLearn](https://github.com/helgoboss/realearn)). For the same reason I actually want to integrate the newtype name in the error message, but that's stuff for another PR. So what do you think about something like this (untested code)?

```rust
#[cfg(feature = "std")]
{
    assert!($name::is_valid(value), format!("{} is not a valid value", value));
}
#[cfg(feature = "no_std")]
{
    assert!($name::is_valid(value), "Not a valid value");
}
```

Reply:
Indeed you can have special case for `std`/`no_std`. In the tests I kept all the other dependencies to `std`, even though it might be possible to make also test working in `no_std` I don't think there is much benefit from it.

And as you mention, if there is a `try_into()` available then I have no issue with `new` panic`:ing.

I haven't yet tested the `helgoboss-midi` in a real application, I'll have a go at that later tonight integrating it in a crate `usbd-hid-midi` that I'm working on.

On another note, your use case is VST plugins and such. I'm also interested in that, have been playing around with JUCE a bit just to get a feel of the existing VST ecosystem. There are now some progress also on `baseview`/`baseplug` with `iced` based GUI bindings. I was looking into using `druid` but it turned out complicated. I have never coded for Reaper, but just saw that there is Rust bindings in the making. Very interesting! 

/Per